python/ray/data/tests/test_huggingface.py

```diff
@@ -46,8 +46,18 @@ def test_huggingface(ray_start_regular_shared):
     reason="IterableDataset.iter() added in 2.8.0",
 )
 def test_from_huggingface_streaming(ray_start_regular_shared):
-    hfds = datasets.load_dataset("tweet_eval", "emotion", streaming=True, split="train")
+    # Returns pyarrow.Table batches.
+    hfds = datasets.load_dataset(
+        "tweet_eval", "emotion", streaming=True, split="train"
+    ).with_format("arrow")
+    assert isinstance(hfds, datasets.IterableDataset)
+    ds = ray.data.from_huggingface(hfds)
+    assert ds.count() == 3257
```

Comment:
Could we parameterize this test and make sure it works with all of the formats? We can avoid the code duplication, and also make sure it works with datasets that don't return `Table` or `dict` batches

```python
@pytest.mark.parametrize("format", [None, 'numpy', 'torch', 'tensorflow', 'pandas', 'arrow', 'jax'])
def test_from_huggingface_streaming(format, ...):
    ...
    ).with_format(format)
    ...